src/Compilers/VisualBasic/Portable/Symbols/Source/SourceMemberContainerTypeSymbol.vb

```diff
@@ -3175,7 +3175,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
         End Function
 
         Friend Overrides Function GetSimpleNonTypeMembers(name As String) As ImmutableArray(Of Symbol)
-            If _lazyMembersAndInitializers IsNot Nothing OrElse MemberNames.Contains(name) Then
+            If _lazyMembersAndInitializers IsNot Nothing OrElse MemberNames.Contains(name, IdentifierComparison.Comparer) Then
```

Comment:
@dotnet/roslyn-compiler this was an intentional change i made as i'm fairly certain the original code is buggy (and can cause race conditions).  I discovered this because one of my tests (TestSafeWithMatchingExtensionMethod_CaseInsensitive) was failing.  Debugging through led me to this line.

In that test, we use the semantic model and try to bind `42.m` where `M` is an actual defined extensino method on 42.  This extension method is *not* found due to the `.Contains` check above failing (since it's case sensitive Contains).  The problem here is that `MemberNames` is just an `IEnumerable<string>`, so we don't get normal VB sensitivity rules working.

Feel free to try to repro this using this PR at commit 0aaa481fabb56bfb30a334fb0817605b1e28e61d.  

My guess is that this doesn't fire often because more often than not `_lazyMembersAndInitializers` is not `nothing` and so we do the normal `GetMembers(name)` call.